lib/OrdinaryDiffEqNonlinearSolve/src/newton.jl

```diff
@@ -529,8 +529,21 @@ function Base.resize!(nlcache::NLNewtonCache, ::AbstractNLSolver, integrator, i:
     resize!(nlcache.atmp, i)
     resize!(nlcache.dz, i)
     resize!(nlcache.du1, i)
-    if nlcache.jac_config !== nothing
-        resize_jac_config!(nlcache.jac_config, i)
+    if !isnothing(nlcache.jac_config) && !isnothing(nlcache.jac_config[1])
+        uf = nlcache.uf
+        uf = SciMLBase.@set uf.f = SciMLBase.unwrapped_f(uf.f)
+
+        # for correct FiniteDiff dirs
+        autodiff_alg = alg_autodiff(integrator.alg)
+        if autodiff_alg isa AutoFiniteDiff
+            ad_right = SciMLBase.@set autodiff_alg.dir = 1 
+            ad_left = SciMLBase.@set autodiff_alg.dir = -1
```

Comment:
I just realized that the default `dir` in `AutoFiniteDiff` is true. What happens when you call `@set` with a negative integer? Does it work? In any case, the type of the `ad` object itself will change, so beware of that.

Reply:
I think the reasoning behind `true` is that it is the "smallest" possible `1`, in terms of type constraints.
My remark was just meant to emphasize that `typeof(AutoFiniteDiff(dir=true)) != typeof(AutoFiniteDiff(dir=-1))`.